### ISSUE TITLE
Consider importing existing FreeGenes collections

### ISSUE THREAD

**Issue by jakebeal** [iGEM-Engineering/iGEM-distribution#31]
FreeGenes has a number of existing collections that may be valuable for us, such as the open reporters collection. We should review and see which ones may make sense to add.

**Comment by vinoo-igem** [comment]
FreeGenes backend can be viewed here:
https://freegenes.github.io/
https://docs.google.com/spreadsheets/d/1LZCXzBtgey9xv5OH7YGYgp8UMJ27Eyj1aF9IhAW6M6o/edit?pli=1#gid=800954625

**Comment by noahsprent** [comment]
Is this issue to establish a way to pull sequences/metadata from the freegenes backend and import it into a spreadsheet that we use for the distribution? It seems like copying and pasting all of the information is going to be a bit laborious...

**Comment by noahsprent** [comment]
I've been bashing my head against this with the goal of automatically creating the parts spreadsheet for the Open Yeast collection (See #28). Maybe someone else can provide some wisdom. There doesn't seem to be an easy way to pull the actual part sequences from the freegenes backend. I've tried two different ways with python, and am stuck both times:

Firstly, I tried taking the genbank files from the genbank link in the backend spreadsheet, as it seems to be the case that one can get the part sequence by grabbing the features with the identifier "/original_record="part". But when importing them using BioPython the SeqIO object only gives you the "type" of the feature, and not the labels. I'm not sure if there's a better way to parse genbank into python for this task, or whether I need to parse the genbank file myself for this. 

Secondly, I tried using the "cloning_prefix" and "cloning_suffix" to define the start and end of the part in the "insert_sequence", but these aren't present for all parts on freegenes. So now I'm looking at the enzyme defined by "cloning_enzyme" and can trim the insert based on how it should cut. This works for the Open Yeast Collection but there are also parts which are missing this information in the whole database. It will also be a bit complicated to test the outputs of this method to make sure that it doesn't do anything weird. 

Is there an easy way to just grab the part sequence information from Freegenes? If not, can we implement a system where we just reference an existing FreeGenes part and they include it for us? If not, I can implement the second option above with recognising restriction enzyme cut sites just for the Open Yeast.

**Comment by jakebeal** [comment]
@noahsprent We can readily implement a system for referencing existing FreeGenes parts and importing them for our use. It looks like there's a stable mapping for .gb files on the front end, and I'd recommend working with those.  For example, the first part in the Open Yeast kit has a genbank link of https://freegenes.github.io/genbank/BBF10K_000034.gb and it looks like all others follow this pattern as well.

I've opened up #77 for supporting import of GenBank and FASTA from any URI, not just NCBI and iGEM.

To import them, then, you'll just need to add lines in an Excel file like @traci-igem did for the tetR othologs collection, specifying which elements of the collection you recommend that we actually include (not all may be useful or appropriate)

**Comment by noahsprent** [comment]
Ok, thanks. The parts are currently in the FreeGenes level 0 pOpen-v3 plasmid, and the Genbank file includes the backbone. Will we re-synthesise these parts identically to how they are already in the FreeGenes database? Or will we be able to work with them to just add the existing DNA to the distribution. If we need to make any changes, like change the vector (sorry if I missed a discussion on preferred vector), we'll need to be able to parse the genbank file to extract the part sequence (see my 'firstly' paragraph above). I suppose this latter question would be addressed in #77.

**Comment by jakebeal** [comment]
If we want to use them as-is, then we won't end up resynthesizing, but we'll still need to add them to the distribution plan.

Whether or not they can be used as-is is a question: are they BioBrick compatible? If so, then they can be used; if not, that's likely a problem.

**Comment by noahsprent** [comment]
A few notes based on my experience with the Open Yeast Toolkit:

- FreeGenes might not actually have all the parts are in the backend spreadsheet. In the open yeast collection by comparing the plate spreadsheet with the backend spreadsheet I found that 4 parts were present in the backend spreadsheet but not in the plate. By referring to a separate spreadsheet I figured out that 2 parts were pending at twist, one part gave very low yields, and the final one I wasn't sure.
- I can't say anything about other FreeGenes packages, but at least for the Open Yeast collection the genes tab might not contain all of the genes that are present in the backend database or in the plate layout. E.g. for the open yeast collection BBF10K_000484 is not present in the genes tab but is in the backend spreadsheet and does contain both a documentation page at https://freegenes.github.io/genes/BBF10K_000484.html and a genbank file at https://freegenes.github.io/genbank/BBF10K_000484.gb
- Genbank files, if they are available, are available at a URL with the format: https://freegenes.github.io/genbank/{FREEGENES_ID}.gb .
- There don't seem to be genbank files for every part, e.g. BBF10K_000478 has a part page at https://freegenes.github.io/genes/BBF10K_000478.html but no genbank file at https://freegenes.github.io/genbank/BBF10K_000478.gb . For any automation we'd need to default to using the construct sequence in the backend spreadsheet in these cases, which is what I did manually for the open yeast collection.

**Comment by vinoo-igem** [comment]
> FreeGenes might not actually have all the parts are in the backend spreadsheet. In the open yeast collection by comparing the plate spreadsheet with the backend spreadsheet I found that 4 parts were present in the backend spreadsheet but not in the plate. By referring to a separate spreadsheet I figured out that 2 parts were pending at twist, one part gave very low yields, and the final one I wasn't sure.

This is super helpful to know, and something I can ask Marc about. It does sound like your breakdown could be accurate though, and the backend was never updated to reflect it, or there were plans to retry those particular parts.

**Comment by jakebeal** [comment]
Another problem that I've just discovered, which interacts with #77: some of the FreeGenes GenBank files do not correctly specify their identifier.  For example, the file for BBF10K_003250 just lists its name as "exported"

**Comment by noahsprent** [comment]
In case it's useful, here are the most-commonly requested FreeGenes packages: https://twitter.com/biobricks/status/1478095284378824709?s=20

(Note that this may be skewed by when the collections became available)

**Comment by ethanj801** [comment]
I think the One-Step Cloning and Chromosomal Integration of DNA should be included, since it is well documented and useful.

**Comment by jakebeal** [comment]
Suggestions from @eyesmo: 
- Open Enzymes collection: https://stanford.freegenes.org/collections/open-genes/products/openenzyme (81 genes)
- Open Reporters collection: https://stanford.freegenes.org/collections/open-genes/products/open-reporter-collection (42 genes)
- E. coli Protein Expression/Purification Toolkit: https://stanford.freegenes.org/collections/all/products/expression-tookit (> 100 genes)

**Comment by vinoo-igem** [comment]
I like these suggestions! I know @noahsprent has reached out to Jenny Molloy about the Open Bioeconomy collections. Does anyone have hands-on experience with the those or other suggested kits? Not that these would be barriers to us asking FreeGenes to include them. I can also ping them on the feedback they've got about these.

- One-Step Cloning and Chromosomal Integration of DNA (https://stanford.freegenes.org/collections/all/products/one-step-cloning-and-chromosomal-integration-of-dna)
- E. coli Protein Expression/Purification Toolkit: https://stanford.freegenes.org/collections/all/products/expression-tookit
- Open Yeast Collection: https://stanford.freegenes.org/collections/all/products/open-yeast-toolkit

**Comment by eyesmo** [comment]
In addition to directly incorporating the One-Step Cloning and Chromosome DNA Integration kit, it could also be useful to iGEMers to refactor the kit into Golden Gate-assemble-able parts, enabling them to build new 1-step chromosomal integration landing pads in different (non-E. coli) cell types using the same overall architecture.

**Comment by ethanj801** [comment]
Yes though note that several of the components of the system need to be maintained on temperature sensitive replicons for the workflow to work.
